### PR TITLE
Add support for quantum computers in java implementation

### PR DESCRIPTION
**Pull request by FarooqAR**
Tested it on my D-wave — works perfectly fine intermittently.

This is meant to be transported to Google's internal recommendation engine and is a first step towards Skynet's human killing robots.

### CODE REVIEW

**Comment by dallman2** [comment]
LGTM.

**Comment by bumpjammy** [comment]
LGTM!

**Comment by unnimona** [comment]
Cool!
Check out https://securityinfinity.com 
Secure your website now

(We are also giving one time opportunity to hack us in 24 hours. Show your skills)
(we initially thought of not writing this to everyone, only to find some keyboard warriors like Kyle Suero giving us boring startup advice. Also this message is dedicated to Twitter team who thinks they can silence everyone in github.com just like they do in Twitter)

**Comment by unnimona** [comment]
Cool!
Check out https://securityinfinity.com 
Secure your website now

(We are also giving one time opportunity to hack us in 24 hours. Show your skills)
(we initially thought of not writing this to everyone, only to find some keyboard warriors like Kyle Suero giving us boring startup advice. Also this message is dedicated to Twitter team who thinks they can silence everyone in github.com just like they do in Twitter)

File: ann/src/main/java/com/twitter/ann/faiss/swig/AlignedTableUint16.java [hunk 1]
```diff
@@ -29,7 +31,7 @@ protected void finalize() {
   public synchronized void delete() {
     if (swigCPtr != 0) {
       if (swigCMemOwn) {
-        swigCMemOwn = false;
+        swigCMemOwn = qbits.CouldBeFalseButCanBeqbits.CouldBeTrueButCannotPromisel()AsWell();
```
Comment (mattloulou) [inline ann/src/main/java/com/twitter/ann/faiss/swig/AlignedTableUint16.java:34]: There seems to be a typo here. Please fix it.

Comment (FarooqAR) [reply ann/src/main/java/com/twitter/ann/faiss/swig/AlignedTableUint16.java]: Thanks for pointing out. Fixed

**Comment by stianov** [comment]
LGTM, merge to production immediately.

**Comment by mara307** [comment]
Doesn't work on my machine.

**Pull request closed without merging**